Kinvey.Tests/DataStoreTests/DataStoreAutoIntegrationTests.cs

```diff
@@ -5627,189 +5789,261 @@ public async Task TestSaveAsync()
-            var existingToDos = await syncToDoStore.FindAsync();
+            var existingItemCache = await syncStore.FindByIDAsync(savedItem.ID);
+            var existingItemNetwork = await networkStore.FindByIDAsync(savedItem.ID);
 
-            // Teardown
-            await autoToDoStore.RemoveAsync(savedToDos.Entities[0].ID);
-            await autoToDoStore.RemoveAsync(savedToDos.Entities[1].ID);
-            await autoToDoStore.RemoveAsync(savedToDos.Entities[2].ID);
-            await autoToDoStore.RemoveAsync(savedToDos.Entities[3].ID);
 
-            // Assert
-            Assert.AreEqual(4, savedToDos.Entities.Count);
+            //Teardown
+            await networkStore.RemoveAsync(savedItem.ID);
+
+            // Assert
+            Assert.IsNotNull(savedItem);
+            Assert.AreEqual(newItem.ID, savedItem.ID);
+            Assert.AreEqual(newItem.Name, savedItem.Name);
+            Assert.AreEqual(newItem.Details, savedItem.Details);
+            Assert.AreEqual(newItem.DueDate, savedItem.DueDate);
+            Assert.IsNotNull(existingItemCache);
+            Assert.AreEqual(newItem.Name, existingItemCache.Name);
+            Assert.AreEqual(newItem.Details, existingItemCache.Details);
+            Assert.AreEqual(newItem.DueDate, existingItemCache.DueDate);
+            Assert.IsNull(existingItemCache.Acl);
+            Assert.IsNull(existingItemCache.Kmd);
+            Assert.IsNotNull(existingItemNetwork);
+            Assert.AreEqual(newItem.ID, existingItemNetwork.ID);
+            Assert.AreEqual(newItem.Name, existingItemNetwork.Name);
+            Assert.AreEqual(newItem.Details, existingItemNetwork.Details);
+            Assert.AreEqual(newItem.DueDate, existingItemNetwork.DueDate);
+        }
+
+        [TestMethod]
+        public async Task TestSaveUpdateVersion5ConnectionIssueAsync()
+        {
+            // Setup
+            kinveyClient = BuildClient("5");
+
+            if (MockData)
+            {
+                MockResponses(1);
+            }
+
+            // Arrange
+            var autoStore = DataStore<ToDo>.Collection(toDosCollection, DataStoreType.AUTO);
+            var syncStore = DataStore<ToDo>.Collection(toDosCollection, DataStoreType.SYNC);
+            var networkStore = DataStore<ToDo>.Collection(toDosCollection, DataStoreType.NETWORK);
+
+            var newItem = new ToDo
+            {
+                ID = Guid.NewGuid().ToString(),
+                Name = "todo save",
+                Details = "details for save",
+                DueDate = "2016-04-22T19:56:00.961Z"
+            };
+
+            await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
+
+            // Act
+            SetRootUrlToKinveyClient(unreachableUrl);
+            var savedItem = await autoStore.SaveAsync(newItem);
+            SetRootUrlToKinveyClient(kinveyUrl);
+
+            var existingItemCache = await syncStore.FindByIDAsync(savedItem.ID);
```

Comment:
I think it would be better to use just Find and assert the number of items - if for some reason two item is recorded multiple times

Reply:
Ok.